src/Tasks/ManifestUtil/SecurityUtil.cs

```diff
@@ -514,8 +534,18 @@ public static void SignFile(string certThumbprint, Uri timestampUrl, string path
                     throw new ArgumentException("TargetFrameworkVersion");
                 }
 
-                // SHA-256 digest can be parsed only with .NET 4.5 or higher.
-                bool isTargetFrameworkSha256Supported = targetVersion.CompareTo(s_dotNet45Version) >= 0;
+                bool isTargetFrameworkSha256Supported = false;
+                if (String.IsNullOrEmpty(targetFrameworkIdentifier) ||
+                    targetFrameworkIdentifier.Equals(Constants.DotNetFrameworkIdentifier, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // SHA-256 digest can be parsed only with .NET 4.5 or higher.
+                    isTargetFrameworkSha256Supported = targetVersion.CompareTo(s_dotNet45Version) >= 0;
+                }
+                else if (targetFrameworkIdentifier.Equals(Constants.DotNetCoreAppIdentifier, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    // Use SHA-256 digest for .NET Core apps
+                    isTargetFrameworkSha256Supported = true;
+                }
```

Comment:
nit:
```suggestion
                // Use SHA-256 digest for .NET Core apps and .NET Framework 4.5 or higher.
                bool isTargetFrameworkSha256Supported =
                targetFrameworkIdentifier.Equals(Constants.DotNetCoreAppIdentifier, StringComparison.OrdinalIgnoreCase) ||
                (targetFrameworkIdentifier.Equals(Constants.DotNetFrameworkIdentifier, StringComparison.OrdinalIgnoreCase) &&
                targetVersion.CompareTo(s_dotNet45Version) >= 0);
```
?

Reply:
Thanks for the suggestion. I prefer to keep it this way for readability reasons.
For .NET 5, the identifier will remain as .NetCoreApp so it should work fine as far as I can tell now.